presto-main/src/main/java/com/facebook/presto/tdigest/TDigest.java

```diff
@@ -238,6 +248,12 @@ private void mergeNewValues(boolean force, double compression)
         }
 
         if (force || unmergedWeight > 0) {
+            // ensure buffer is big enough to hold current centroids and incoming values
+            while (activeCentroids + tempUsed >= tempMean.length) {
```

Comment:
Sorry I'm not familiar with the algorithm. But can you just compute how much needs to be copied and copy once?